packages/mobile/src/screens/chat-screen/ChatListItem.tsx

```diff
@@ -103,38 +111,38 @@ export const ChatListItem = ({ chatId }: { chatId: string }) => {
 
   return (
     <TouchableHighlight onPress={handlePress}>
-      <View style={styles.root}>
-        {otherUser ? (
-          <>
-            <View style={styles.contentRoot}>
-              <View style={styles.userContainer}>
-                <ProfilePicture
-                  profile={otherUser}
-                  style={styles.profilePicture}
-                />
-                <View style={styles.userTextContainer}>
-                  <View style={styles.userNameContainer}>
-                    <Text style={styles.userName}>{otherUser.name}</Text>
-                    <UserBadges user={otherUser} hideName />
-                  </View>
-                  <Text style={styles.handle}>@{otherUser.handle}</Text>
+      {otherUser ? (
```

Comment:
feels like this should be checking for `!!chat`, but I guess it's helpful to assert otherUser is defined

Reply:
Even if `chat` is defined, we still need to wait to load in the user details.
However I need to rewrite this a bit anyway, stay tuned.

Reply:
What does this look like if `otherUser` exists but `chat` doesn't?

Reply:
this won't get rendered - @dharit-tan and I chatted in slack a bit about this, it's a bit wonky

- Right now in fetching chats we trigger fetching the appropriate users, but don't wait for that to complete
- The way this PR is now is only waiting on that second _users_ fetching, not the original chats loading (which if we have no chats rn it shows empty state I believe?)

This presents a problem because if the chats aren't loaded we want the fading skeletons, but what do we do if the users aren't loaded? We could check to make sure all users are loaded, but then if you paginate, the whole chat list is going be replaced with skeletons until the new users appear

We have a couple options:
- Show the fading skeletons, and make the saga wait for all the relevant users to be fetched before returning so we can switch straight from the fading skeletons to complete chat list items
- Show the fading skeletons, then when we have chats, show non-fading skeletons (one for each chat), then when the users are loaded, show the final chat list items

I think the former is better, @dharit-tan is looking into it

Reply:
Agree option 1 is better, but the fns for retrieving users is in web common and I couldn't figure out how to get them into actual common.

Implemented option 2 here, don't love how it looks in the code but I believe it works.

Reply:
That checks out - I think that's why I didn't do option 1 initially 🙁 

will take a look at the option 2 impl